Libraries/LibWeb/HTML/CookieStore.idl

```diff
@@ -0,0 +1,106 @@
+#import <DOM/Event.idl>
+#import <DOM/EventHandler.idl>
+#import <DOM/EventTarget.idl>
+#import <HighResolutionTime/DOMHighResTimeStamp.idl>
+
+// https://wicg.github.io/cookie-store/
+dictionary CookieStoreGetOptions {
+    USVString name;
+    USVString url;
+};
+
+enum CookieSameSite {
+    "strict",
+    "lax",
+    "none"
+};
+
+dictionary CookieInit {
+    required USVString name;
+    required USVString value;
+    DOMHighResTimeStamp? expires = null;
+    USVString? domain = null;
+    USVString path = "/";
+    CookieSameSite sameSite = "strict";
+    boolean partitioned = false;
+};
+
+dictionary CookieChangeEventInit : EventInit {
+    CookieList changed;
+    CookieList deleted;
+};
+
+dictionary CookieStoreDeleteOptions {
+    required USVString name;
+    USVString? domain = null;
+    USVString path = "/";
+    boolean partitioned = false;
+};
+
+dictionary CookieListItem {
+    USVString name;
+    USVString value;
+    USVString? domain;
+    USVString path;
+    DOMHighResTimeStamp? expires;
+    boolean secure;
+    CookieSameSite sameSite;
+    boolean partitioned;
+};
+
+typedef sequence<CookieListItem> CookieList;
+
+[Exposed=(ServiceWorker,Window), SecureContext]
+interface CookieStore : EventTarget {
+//  Promise<CookieListItem?> get(USVString name);
+//  Promise<CookieListItem?> get(optional CookieStoreGetOptions options = {});
+
+//  Promise<CookieList> getAll(USVString name);
+//  Promise<CookieList> getAll(optional CookieStoreGetOptions options = {});
+
+//  Promise<undefined> set(USVString name, USVString value);
+//  Promise<undefined> set(CookieInit options);
+
+//  Promise<undefined> delete(USVString name);
+//  Promise<undefined> delete(CookieStoreDeleteOptions options);
+
+//  [Exposed=Window]
+//  attribute EventHandler onchange;
```

Comment:
Rather than commenting out unimplemented methods and attributes, you can annotate them with the `[FIXME]` extended attribute.